vpr/src/route/partition_tree.cpp

```diff
@@ -139,20 +149,65 @@ std::unique_ptr<PartitionTreeNode> PartitionTree::build_helper(const Netlist<>&
         for (auto net_id : nets) {
             t_bb bb = route_ctx.route_bb[net_id];
             if (bb.ymax < best_pos) {
-                left_nets.push_back(net_id);
+                left_nets.insert(net_id);
             } else if (bb.ymin > best_pos) {
-                right_nets.push_back(net_id);
+                right_nets.insert(net_id);
             } else {
-                my_nets.push_back(net_id);
+                my_nets.insert(net_id);
             }
         }
 
         out->left = build_helper(netlist, left_nets, x1, y1, x2, std::floor(best_pos));
         out->right = build_helper(netlist, right_nets, x1, std::floor(best_pos + 1), x2, y2);
     }
 
+    if(out->left)
+        out->left->parent = out.get();
+    if(out->right)
+        out->right->parent = out.get();
+
+    out->bb = {x1, x2, y1, y2, 0, 0};
```

Comment:
Should comment that you are assuming a one layer FPGA.  Might be better to use 0, grid->max_layers-1.  That would turn the bb off in the 3rd dimension, which should work I think ...